lib/msal-browser/docs/events.md

```diff
@@ -83,7 +99,10 @@ const callbackId = msalInstance.addEventCallback((message: EventMessage) => {
 
 ## Syncing logged in state across tabs and windows
 
-If you would like to update your UI when a user logs in or out of your app in a different tab or window you can subscribe to the `ACCOUNT_ADDED` and `ACCOUNT_REMOVED` events. The payload will be the `AccountInfo` object that was added or removed.
+If you would like to update your UI when a user logs in or out of your app or changes the active account in a different tab or window you can subscribe to the `ACCOUNT_ADDED`, `ACCOUNT_REMOVED`, and `ACTIVE_ACCOUNT_UPDATED` events.
+
+-   For account additions and removals, the payload will be the `AccountInfo` object that was added or removed.
+-   For active account updates, the payload will contain the previous and new active account filters as `AccountFilter` objects.
```

Comment:
Why filters? Why not the AccountInfo object here as well?

Reply:
@tnorling the storage event is activated by the filters changing, there isn't necessarily a change at the cached account level given that active account can now change due to changing the active tenant profile (hence the same account object in the cache, only changing what gets returned by getActiveAccount()). The account objects aren't recorded in the storage event but the old/new filters are.

Returning the account objects would require reading them from the cache (calling getAccount for each). The calling app can still pass the filters directly to getAccount to get the exact accounts in question if they want to, I'd rather we didn't force the perf hit on them.

Reply:
What happens if we change the way we store/identify active accounts? It sounds like we're leaking implementation details here. I don't disagree r.e. perf but given this is a public interface change we should first consider what we want the public interface to be then figure out how to build it, rather than defining the interface based on today's implementation, if that makes sense.

Personally I think returning the AccountInfo object is both more intuitive and consistent with the rest of our APIs. If you disagree or if doing so results in perf concerns that perhaps warrants further discussion.

Reply:
FWIW cache lookups should be inexpensive. If they're taking double digit or more ms that's something we should look into separately

Reply:
Here's my take:

Semantically, `ACTIVE_ACCOUNT_CHANGED` is an event that signals a contextual/app state change across browser tabs/windows. The `AccountInfo` object is orthogonal given that it suffers no changes to cause or as a result of this storage event.

The actual object that changed in the cache is the filters object, which is both 
a) the minimum required context that the subscriber really needs to react to the event (e.g. update UI in a different tab to maintain consistency across tabs) and
b) the only context we can source from the storage event itself without additional cache lookups/parsing.

We can certainly re-design how we keep track of the active account, but I think that will also open us up to backward compatibility challenges and duplication of account info in the cache.

Happy to draft a design doc and set up some for the team to discuss this week, I agree we may benefit from further discussion.

Reply:
Based on my current knowledge of MSAL I'm not actually sure what `AccountFilter` is, how the app would use it, or how it impacts the frequency at which this is called. Superficially (ignoring architectural implications) it does seem more consistent to pass in `AccountInfo` as `ACCOUNT_ADDED` and `ACCOUNT_REMOVED` do.

In terms of triggering, what are the situations in which this new event would get called? Whenever `tenantId` and `uniqueId` change per [Accounts in MSAL Javascript](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/docs/Accounts.md#homeaccountid)? (I guess based on the test I see `homeAccountId`, `localAccountId` and `tenantId`, but not sure if there is any overlap here.) I think it might be useful to enumerate these conditions in the API event docs too.

Reply:
@JasonGore `AccountFilter` is an extended subset of `AccountInfo` that is used internally for cache lookups and is exposed publically so the apps can pass it into `getAccountX` APIs. The specific fitler objects published in this event would contain the home/local account IDs and tenant ID (of previous and new active accounts).

So the way you'd use it in this case:
- To have context of what account was active before the event
- To get the new active account by passing in the newActiveAccountFilter into `getAccount` (alternatively and perhaps more correctly you could just call `getActiveAccount` and update your app state.

W.r.t. triggering, in the MSAL context, this event is triggered in subscribed tabs when `setActiveAccount` is called in any tab. In actual code, MSAL will emit the event whenever the cache entry for your clientId ending in ".active-account-filters" is modified in any way.